CMakeLists.txt

```diff
@@ -132,6 +132,13 @@ set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE C)
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH})
 set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION ${VERSION_MAJOR})
 
+if(CMAKE_COMPILER_IS_GNUCC)
```

Comment:
Nit: I think this variable is deprecated, so could we use `CMAKE_C_COMPILER_ID` instead? https://cmake.org/cmake/help/latest/variable/CMAKE_COMPILER_IS_GNUCC.html

Reply:
Nit 2: Could we add a comment about why we disable the static analyzer on version less than 10? E.g. "static analyzer wasn't available in GCC until the 10.0 release" or something like that.